ts/lib/auth/googleauth.ts

```diff
@@ -353,8 +362,8 @@ export default class GoogleAuth {
    * @param {object=} json The input object.
    * @param {function=} opt_callback Optional callback.
    */
-  public fromJSON(json, opt_callback) {
-    let client = null;
+  public fromJSON(json: any, opt_callback: (err: Error, client?: any) => void) {
```

Comment:
Something like:
```
interface Json {
    [key: string]: string | number | boolean | Json;
}
```
isn't perfect, but using that or `string` could help distinguish if the type should be the string representation of the JSON or the JSON string decoded as an object.